### DISCUSSION TITLE
Attribute names change between 1.21 and 1.22?

### DISCUSSION THREAD

**Discussion by grafalex82** [Koenkk/zigbee2mqtt#10423] [Devices]
My [external converter](https://github.com/grafalex82/hellozigbee/blob/main/myswitch.js) stopped working some time between 1.21.1 and 1.22.1, and I am trying to understand why.

The device exposes a few endpoints, with some settings attributes:
```
function genSwitchEndpoint(epName) {
    return [
        e.switch().withEndpoint(epName),
        exposes.enum('switch_mode', ea.ALL, switchModeValues).withEndpoint(epName),
        exposes.enum('switch_actions', ea.ALL, switchActionValues).withEndpoint(epName),
        exposes.enum('relay_mode', ea.ALL, relayModeValues).withEndpoint(epName),
        exposes.numeric('max_pause', ea.ALL).withEndpoint(epName),
        exposes.numeric('min_long_press', ea.ALL).withEndpoint(epName),
    ]
}
```

When I set an attribute from the dashboard in z2m 1.21.1 I see this message:
```
Received MQTT message on 'zigbee2mqtt2/TestSwitch/button_2/set' with data '{"switch_mode":"multifunction"}'
```
The endpoint name (button_2) is in topic name, while the payload has only attribute name and value.

In 1.22.1 behavior changed:
```
Received MQTT message on 'zigbee2mqtt2/TestSwitch/button_2/set' with data '{"switch_mode_button_2":"multifunction"}'
```
Now endpoint name is added to the attribute name. This causes an error like this:
```
No converter available for 'switch_mode_button' (undefined)
```

It is not clear at the moment whether this is a regression, or expected new behavior. I tend to think this is a bug, since attribute name gets corrupted - only a part of the endpoint name was truncated.

Unfortunatley I do not see a way how to properly register a converter. Does anyone have an opinion on this?

**Comment by Koenkk** [comment]
@nurikk could you check this? When using the `property` the command should be send to the non-endpoint topic, `zigbee2mqtt2/TestSwitch/set` in this example

**Comment by nurikk** [comment]
I can check only after 2 weeks

**Comment by Koenkk** [comment]
Issue should be fixed now in the latest dev

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by grafalex82** [reply]
Just checked. Thank you, it is working now!
